src/languages/php.js

```diff
@@ -11,23 +11,101 @@ Category: common
  * @returns {LanguageDetail}
  * */
 export default function(hljs) {
+  const regex = hljs.regex;
   const VARIABLE = {
-    className: 'variable',
+    scope: 'variable',
     begin: '\\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
       // negative look-ahead tries to avoid matching patterns that are not
       // Perl at all like $ident$, @ident@, etc.
       `(?![A-Za-z0-9])(?![$])`
   };
+  const CONSTANT = {
+    variants: [
+      {
+        begin: [
+          /const\s+/,
+          /\w+/,
+          /\s*=/,
+        ],
+        beginScope: {
+          1: "keyword",
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          regex.concat(
+            /::/,
+            regex.lookahead(/(?!class\b)/)
+          ),
+          /(\w+\b(?!\())/
+        ],
+        beginScope: {
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          regex.concat(
+            /[A-Z]\w*/,
+            /::/,
+            regex.lookahead(/(?!class\b)/)
+          ),
+          /(\w+\b(?!\())/,
+        ],
+        beginScope: {
+          1: "title.class",
+          2: "variable",
+        },
+      },
+      {
+        begin: [
+          /::/,
+          /class/,
+        ],
+        beginScope: {
+          2: "keyword",
+        },
+      },
+      {
+        begin: [
+          /[A-Z]\w*/,
```

Comment:
Lets wrap this in a constant since it's re-used multiple times.

Reply:
1.
Because now I have an error, I found the root of my regression. The match is case-insensitive because it uses `i` by default. It totally ignores the uppercase rule. Can I disable it or is it some kind of bug?
``` 
Error: php/[9]/begin: Polynomial backtracking. By repeating any character that matches /\xa0/i, an attack string can be created.

    \s+)([a-zA-Z_-ÿ][a-zA-Z0-9_-ÿ]*
    ^~~              ^~~~~~~~~~~~~~~~

Full pattern:
/(const)(\s+)([a-zA-Z_-ÿ][a-zA-Z0-9_-ÿ]*(?![A-Za-z0-9])(?![$]))(\s*=)/im
```

2. I also added a new const for name and in MR description added info about names.

Reply:
@joshgoebel Do you know anything about this case-insensitive matching?

Reply:
PHP is `case_insensitive: true,` so EVERY regex is `/i`, there i no need to specify and it's impossible to have ANY regex that is case-dependent.

So forget about using `CamelCase` rules... I've read not ALL of PHP is case-sensitive though... perhaps we could switch it to "case-sensitive" and try and deal with the exceptions by hand?  How familiar are you with PHP's case sensitivity overall?

Reply:
> PHP is `case_insensitive: true,` so EVERY regex is `/i`, there i no need to specify and it's impossible to have ANY regex that is case-dependent. I forgot this or I'd have mentioned it earlier.

Now that's news.  How could I miss this `case_insensitive: true,` :)

I checked the current implementation of the PHP highlighter and part of camel case keywords are affected by this. I would separate keywords into case-insensitive (false, if)  and case-sensitive (ArithmeticError ArrayIterator).  Case-insensitive function and class names are not affected by this.

Reply:
>  I would separate keywords into case-insensitive (false, if) and case-sensitive 

Except we have no current way to do that [case insensitive in a case sensitive grammar] - the setting is at the grammar level (because we concat every regex into a single huge one)... I could imagine a possible enhancement to JUST `keywords` to allow "as given" or "all lowercase" (would allow the two most common variants, which might in many cases be sufficient)... but that doesn't necessarily help with any custom rules, which would need every letter individually split out... of course we could potentially write a helper for that...

What do you think of the "as given" + all lowercase... ie, supporting "false" and "FALSE" but not "fAlSe"?

I'm trying to figure out what could be done here to wrap this PR up and then circle back to the larger discussion of casing as a sep concern.